src/components/dumb/TabWorkflow.vue

```diff
@@ -0,0 +1,207 @@
+  id: number;
+  next_states: INextState[];
+  qualifiers: ISaveState[];
+  tag: string;
+}
+
+interface INextState {
+  id: number;
+  qualifiers: { id: number }[];
+}
+
+interface IWorkflowOwner {
+  id: string;
+  type: string;
+  omschrijving: string;
+}
+
+interface IWorkflowState {
+  id: number;
+  comment?: string;
+}
+
+interface IWorkflow {
+  actor: string;
+  datum: string;
+  state: IWorkflowState;
+  owners: IWorkflowOwner[];
+  actor_omschrijving: string;
+}
+
+const props = defineProps<TabWorkflowProps>();
+
+// Grid
+const gridApi = ref<GridApi>();
+const getColumnDefinitions = (): ColDef[] => {
+  return [
+    { headerName: 'Status', field: 'state.id', flex: 1, cellRenderer: statusCellRenderer },
+    { headerName: 'Qualifier', field: 'state.state_qualifier', cellRenderer: qualifierCellRenderer, flex: 1 },
+    { headerName: 'Toegewezen aan', field: 'owners', valueFormatter: ownerValueFormatter, flex: 1 },
+    { headerName: 'Datum aanpassing', field: 'datum', valueFormatter: dateFormatter, flex: 1 },
+    { headerName: 'Aangepast door', field: 'actor_omschrijving', flex: 1 },
+    { headerName: 'Opmerkingen', field: 'state', valueFormatter: remarkValueFormatter, flex: 1 },
+  ];
+};
+
+const statusCellRenderer = (params: ICellRendererParams<number>) => {
+  if (params.value || params.value === 0) {
+    const state = props.workflowSchema.find((s) => s.id === params.value);
+    if (state) {
+      return state.description;
+    }
+    const errorElement = document.createElement('i');
+    errorElement.innerHTML = `Foute state ID: ${params.value}`;
+    errorElement.classList.add('invalid-status');
+    return errorElement;
+  }
+  return '';
+};
+
+const qualifierCellRenderer = (params: ICellRendererParams<IWorkflow, number>) => {
+  if (params.value || params.value === 0) {
+    const state = props.workflowSchema.find((s) => s.id === params.data?.state.id);
+    if (state) {
+      const qualifier = state.qualifiers.find((q) => q.id === params.value);
+      return qualifier?.description || '';
+    }
+    return '<em>foute state ID: ' + params.data?.state.id + '</em>';
+  }
+  return '';
+};
+
+const ownerValueFormatter = (params: ValueFormatterParams<IWorkflow, IWorkflowOwner[]>) => {
+  if (params.value) {
+    params.value = params.value.filter((v) => v.omschrijving);
+    return params.value.map((o) => o.omschrijving).join('; ');
+  }
+
+  return '';
+};
+
+const remarkValueFormatter = (params: ValueFormatterParams<IWorkflowState, IWorkflowState>) => {
+  return params.value?.comment || '-';
+};
+
+const dateFormatter = (params: ValueFormatterParams) => {
+  if (params.value) {
+    return format(new Date(params.value || ''), 'dd-MM-yyyy');
+  }
+  return '';
+};
+
+const gridOptions = ref<GridOptions>({
+  context: getCurrentInstance(),
+  defaultColDef: { sortable: false, resizable: true },
+  suppressMovableColumns: true,
+  suppressClickEdit: true,
+  headerHeight: 45,
+  rowHeight: 40,
+  rowModelType: 'infinite',
```

Comment:
Zijn die tab workflow stuff meestal geen gewone grids ipv infinite?

Reply:
Khad mij gebaseerd op die van proces_energieadvies dus blijkbaar niet

Reply:
Ok maar der wordt gene range ofzo meegegeven dus die infinite gaat ook maar ene call doen voor alle workflow data? In da geval ist imo beter om gewoon de call in app te doen en deze effectief dumb houden en gewoon ne array van workflow data te tonen

Reply:
heb eens bij proces notas gekeken, daar wordt idd maar 1 call gedaan, zonder iets van range headers ofzo